node/src/blockchain/test_utils.rs

```diff
@@ -56,9 +63,139 @@ pub fn make_meaningless_seed() -> Seed {
+        self.transaction_hash_opt = Some(hash);
+        self
+    }
+
+    pub fn transaction_index(mut self, index: Index) -> ReceiptResponseBuilder {
+        self.transaction_index_opt = Some(index);
+        self
+    }
+
+    pub fn block_hash(mut self, hash: Hash) -> ReceiptResponseBuilder {
+        self.block_hash_opt = Some(hash);
+        self
+    }
+
+    pub fn block_number(mut self, number: U64) -> ReceiptResponseBuilder {
+        self.block_number_opt = Some(number);
+        self
+    }
+
+    pub fn cumulative_gas_used(mut self, number: U256) -> ReceiptResponseBuilder {
+        self.cumulative_gas_used_opt = Some(number);
+        self
+    }
+
+    pub fn gas_used(mut self, number: U256) -> ReceiptResponseBuilder {
+        self.gas_used_opt = Some(number);
+        self
+    }
+
+    pub fn contract_address(mut self, hash: H160) -> ReceiptResponseBuilder {
+        self.contract_address_opt = Some(hash);
+        self
+    }
+
+    pub fn logs(mut self, logs: Vec<Log>) -> ReceiptResponseBuilder {
+        self.logs_opt = Some(logs);
+        self
+    }
+
+    pub fn status(mut self, number: U64) -> ReceiptResponseBuilder {
+        self.status_opt = Some(number);
+        self
+    }
+
+    pub fn root(mut self, hash: Hash) -> ReceiptResponseBuilder {
+        self.root_opt = Some(hash);
+        self
+    }
+
+    pub fn logs_bloom(mut self, bloom: H2048) -> ReceiptResponseBuilder {
+        self.logs_bloom_opt = Some(bloom);
+        self
+    }
+
+    pub fn build(self) -> String {
+        let mut transaction_receipt = TransactionReceipt::default();
+
+        if let Some(transaction_hash) = self.transaction_hash_opt {
+            transaction_receipt.transaction_hash = transaction_hash;
+        }
+
+        if let Some(index) = self.transaction_index_opt {
+            transaction_receipt.transaction_index = index;
+        }
+
+        if let Some(cumulative_gas_used) = self.cumulative_gas_used_opt {
+            transaction_receipt.cumulative_gas_used = cumulative_gas_used;
+        }
+
+        if let Some(logs) = self.logs_opt {
+            transaction_receipt.logs = logs;
+        }
+
+        if let Some(bloom) = self.logs_bloom_opt {
+            transaction_receipt.logs_bloom = bloom;
+        }
+
+        transaction_receipt.block_hash = self.block_hash_opt;
+        transaction_receipt.block_number = self.block_number_opt;
+        transaction_receipt.gas_used = self.gas_used_opt;
+        transaction_receipt.contract_address = self.contract_address_opt;
+        transaction_receipt.status = self.status_opt;
+        transaction_receipt.root = self.root_opt;
+
+        let rpc_response = RpcResponse {
+            json_rpc: "2.0".to_string(),
+            id: 0,
+            result: transaction_receipt,
+        };
+        serde_json::to_string(&rpc_response).unwrap()
+    }
+}
+
 #[derive(Default)]
 pub struct BlockchainInterfaceMock {
```

Comment:
I'm sorry for causing troubles but I must state that this is another place where I sense some rough abrasive. This mock was turned into pathetic remains of a useful tool. When you scan over its trait implementation you see all methods not needed except one. That's eyebrow-raising and it indicates something's wrong. 

Usually it simply says that some code was left untested. I know, however, this is rather an effect of your and Utkarsh's experimenting with heavy usages of the MockBlockchainClientServer by whose configuration you try to achieve the tested code, staying in the production version, unaltered, will receive a response identical to a real-world scenario. It's impressive on one hand, but it goes badly together with the standard mocks which we historically have been using a lot. In such a combination, especially if the standard mock is involved significantly partially only, it isn't simply good.

I'd love if we could also look into this more closely with Dan's eyes with us.

Reply:
STATUS: MEDIUM CONCEPTUAL CONCERN. DAN'S COMMENT HIGHLY DESIRABLE. AT LEAST A CARD CLEANING THIS UP SHOULD BE MADE.

Reply:
Hi, I talked a bit on this with Dan and we have this suggestion:
It seems like the desirable thing to do is actually discard this mock, but just ultimately. Before that you'd have to go look at the test using somehow `retrieve_transactions`. It's the only method that might be in use. If you prove true that there is such a test, check again if that test could also be written more in the way you have applied in those others, with the sturdy BlockchainClientMockServer. 

The last thing to consider is how hard it would be to do this extra work within this open card (maybe it wouldn't be). We thought that it might be potentially a good solution for case it's harder, if you wrote a minor card about a clean-up. To rewrite any remaining dependent test and to erase the whole class of this mock off.

We think you have the right knowledge to decide by yourself which path to go.